### PR TITLE
fix some of the warnings.

### PR DESCRIPTION
**Pull request by jennyf19**
Had some issues getting some of them into the config file, so need to do a little more work on some, but started chipping away at the 1000+ errors. Majority now are related to xml + documentation (@jmprieur )

### CODE REVIEW

File: src/Microsoft.Identity.Web/GlobalSuppressions.cs [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+[assembly: SuppressMessage("StyleCop.CSharp.ReadabilityRules", "SA1101:Prefix local calls with this", Justification = "not applicable", Scope = "member", Target = "~P:Microsoft.Identity.Web.TokenCacheProviders.Session.MsalSessionTokenCacheProvider.CurrentHttpContext")]
+[assembly: SuppressMessage("Globalization", "CA1303:Do not pass literals as localized parameters", Justification = "not applicable", Scope = "member", Target = "~M:Microsoft.Identity.Web.Base64UrlHelpers.DecodeToBytes(System.String)~System.Byte[]")]
+[assembly: SuppressMessage("StyleCop.CSharp.OrderingRules", "SA1201:Elements should appear in the correct order", Justification = "not applicable", Scope = "member", Target = "~F:Microsoft.Identity.Web.Resource.OpenIdConnectMiddlewareDiagnostics.s_onRedirectToIdentityProvider")]
+[assembly: SuppressMessage("StyleCop.CSharp.NamingRules", "SA1308:Variable names should not be prefixed", Justification = "not applicable", Scope = "member", Target = "~F:Microsoft.Identity.Web.Resource.OpenIdConnectMiddlewareDiagnostics.s_onRedirectToIdentityProvider")]
```
Comment (pmaytak) [inline src/Microsoft.Identity.Web/GlobalSuppressions.cs:137]: Hmm. Should discuss [SA1101](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1101.md), [SA1201](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1201.md), [SA1308](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1308.md), [CA1303](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1303?view=vs-2019).

File: src/Microsoft.Identity.Web/TokenAcquisition.cs [hunk 2]
```diff
@@ -57,14 +48,26 @@ internal class TokenAcquisition : ITokenAcquisition, ITokenAcquisitionInternal
             _logger = logger;
         }
 
+        private readonly MicrosoftIdentityOptions _microsoftIdentityOptions;
```
Comment (pmaytak) [inline src/Microsoft.Identity.Web/TokenAcquisition.cs:51]: Should discuss this ordering rule.

File: tests/.editorconfig [hunk 3]
```diff
@@ -5,3 +5,39 @@
 
 # CA1707: Identifiers should not contain underscores
 dotnet_diagnostic.CA1707.severity = none
+
```
Comment (pmaytak) [inline tests/.editorconfig:8]: Good that these are in editorconfig. We should discuss some of these.

File: tests/Microsoft.Identity.Web.Test.Integration/GlobalSuppressions.cs [hunk 4]
```diff
@@ -0,0 +1,10 @@
+﻿// This file is used by Code Analysis to maintain SuppressMessage
+// attributes that are applied to this project.
+// Project-level suppressions either have no target or are given
+// a specific target and scoped to a namespace, type, member, etc.
+
+using System.Diagnostics.CodeAnalysis;
+
+[assembly: SuppressMessage("StyleCop.CSharp.ReadabilityRules", "SA1108:Block statements should not contain embedded comments", Justification = "not applicable", Scope = "member", Target = "~M:Microsoft.Identity.Web.Test.Integration.AcquireTokenForAppIntegrationTests.#ctor(Xunit.Abstractions.ITestOutputHelper)")]
+[assembly: SuppressMessage("StyleCop.CSharp.ReadabilityRules", "SA1101:Prefix local calls with this", Justification = "not applicable", Scope = "member", Target = "~M:Microsoft.Identity.Web.Test.Integration.AcquireTokenForAppIntegrationTests.#ctor(Xunit.Abstractions.ITestOutputHelper)")]
+[assembly: SuppressMessage("Design", "CA1062:Validate arguments of public methods", Justification = "not applicable", Scope = "member", Target = "~M:Microsoft.Identity.Web.Test.Integration.AcquireTokenForAppIntegrationTests.#ctor(Xunit.Abstractions.ITestOutputHelper)")]
```
Comment (pmaytak) [inline tests/Microsoft.Identity.Web.Test.Integration/GlobalSuppressions.cs:10]: Yep. Good suppress for tests.

File: tests/Microsoft.Identity.Web.Test/WebApiExtensionsTests.cs [hunk 5]
```diff
@@ -99,27 +108,6 @@ public void AddProtectedWebApi_WithConfigActions(bool useServiceCollectionExtens
             AddProtectedWebApi_TestCommon(services, provider);
         }
 
-        private void AddProtectedWebApi_TestCommon(IServiceCollection services, ServiceProvider provider)
```
Comment (pmaytak) [inline tests/Microsoft.Identity.Web.Test/WebApiExtensionsTests.cs:102]: I think the private methods should follow after the public ones that use them, in this case. Makes it more readable, IMO.

File: tests/Microsoft.Identity.Web.Test/Resource/AadIssuerValidatorTests.cs [hunk 6]
```diff
@@ -1,15 +1,15 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
-using Microsoft.Identity.Web.Resource;
-using Microsoft.IdentityModel.Tokens;
 using System;
 using System.IdentityModel.Tokens.Jwt;
 using System.Security.Claims;
-using Xunit;
+using Microsoft.Identity.Web.Resource;
```
Comment (jmprieur) [inline tests/Microsoft.Identity.Web.Test/Resource/AadIssuerValidatorTests.cs:7]: Is it part of the same PR? (fixing CA issues ?)

File: tests/Microsoft.Identity.Web.Test/ServiceCollectionExtensionsTests.cs [hunk 7]
```diff
@@ -18,15 +18,18 @@ public void AddTokenAcquisition_AddsWithCorrectLifetime()
 
             services.AddTokenAcquisition();
 
-            Assert.Collection(services.OrderBy(s => s.ServiceType.Name),
-                actual => {
+            Assert.Collection(
```
Comment (jmprieur) [inline tests/Microsoft.Identity.Web.Test/ServiceCollectionExtensionsTests.cs:21]: is it part of the same PR (fixing CA issues?)

**Pull request merged**